quarkchain/evm/state.py

```diff
@@ -263,6 +261,7 @@ def __init__(
         self.executing_on_head = executing_on_head
         self.qkc_config = qkc_config
         self.sender_disallow_list = set()  # type: Set[bytes]
+        self.shard_config = ShardConfig(ChainConfig())
```

Comment:
I feel there is too much abstraction leaking from shard to evm state, ideally the evm state should not know what is a shard / chain. how about just pass `default_chain_token` to the constructor?

Reply:
what if we want to add a permitted token list in EVM? how about letting the EVM know which shard it belongs to and what contract from neighbor it can talk to? those are examples of what we can do with a more complete picture of shard config, rather than a default_chain_token

having more info than needed is better considering we might get more features into EVM soon

Reply:
it's no big deal for now. when the config gets more complicated we can refactor to something like `EVMConfig`.

one final note, can you pass the config to the constructor and make it required? otherwise it's possible that the caller forgot to update the `self.shard_config` in place. or make `self.shard_config` to `None` so unexpected invocations will fail immediately

Reply:
we already have quarkchain.evm.config.Env for this purpose

if the caller forgot to update the shard_config, it would use the default value which is the current code, the EVM would have a default value to use

what benefit does failing the invocations have? besides making our untittests fail

Reply:
the hard part is picking an initial/default value for the default token for EVM, using either 0 or "TQKC" is bad. the cleanest way of reusing whatever we defined in ShardState is going to be something like this. I can add an optional argument to `State.__init__` but it does not really help people remember to adjust ShardState, given the argument list is quite long and even contains kwargs